src/node/interfaces.cpp

```diff
@@ -857,16 +861,40 @@ class ChainImpl : public Chain
     NodeContext& m_node;
 };
 
+typedef std::map<CTransactionRef, size_t> TxTemplateMap;
 class BlockTemplateImpl : public BlockTemplate
 {
 public:
-    explicit BlockTemplateImpl(BlockAssembler::Options assemble_options,
-                               std::unique_ptr<CBlockTemplate> block_template,
-                               NodeContext& node) : m_assemble_options(std::move(assemble_options)),
-                                                    m_block_template(std::move(block_template)),
-                                                    m_node(node)
+    explicit BlockTemplateImpl(
+        BlockAssembler::Options assemble_options,
+        std::unique_ptr<CBlockTemplate> block_template,
+        NodeContext& node,
+        std::shared_ptr<TxTemplateMap> tx_template_refs) : m_assemble_options(std::move(assemble_options)),
+                                                          m_block_template(std::move(block_template)),
+                                                          m_node(node),
+                                                          m_tx_template_refs(std::move(tx_template_refs))
+
     {
         assert(m_block_template);
+
+        TxTemplateMap& tx_refs{*Assert(m_tx_template_refs)};
+        // Don't track the dummy coinbase, because it can be modified in-place
+        // by submitSolution()
```

Comment:
b9306b79b8f5667a2679236af8792bb1c36db817: in addition, we might be wiping the dummy coinbase from the template later: https://github.com/Sjors/bitcoin/pull/106